### PR TITLE
Fixes #15212: Generate a file on root server containing all nodes certificate

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15212

### CODE REVIEW

**Comment by fanf** [comment]
We need to trigger a `SIGUP` to `relayd` after writing the file

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/WriteNodeCertificatesPem.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    } yield ()
+  }
+
+  // check that parent directory exists and is writable or try to create it.
+  def checkParentDirOK(file: File): IOResult[Unit] = {
+    val parent = file.parent
+
+    for {
+      _      <- IOResult.effectM(s"Error when trying to create parent directory for node certificate file: ${parent.pathAsString}") {
+                  val p = file.parent
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/WriteNodeCertificatesPem.scala]: why not using parent here ?

Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/WriteNodeCertificatesPem.scala]: I think this should be 
```
IOResult.effectM(s"Error when trying to create parent directory for node certificate file: ${parent.pathAsString}") {
                parent.createDirectoryIfNotExists(true)
}
```

**Comment by fanf** [comment]
PR rebased

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/services/policies/TestWriteNodeCertificatesPem.scala [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+    log.setAdditive(false) // disable root logger/appender
+    log.addAppender(appender)
+
+    // exec
+    writer.writeCerticatesAsync(dest, nodes)
+
+    Thread.sleep(100)
+
+    val output = os.toString(StandardCharsets.UTF_8)
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/services/policies/TestWriteNodeCertificatesPem.scala]: toString(StandardCharsets) is not working in java8 ! 

See https://ci.normation.com/jenkins/job/test-rudder-pr/48/console

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**